.travis.yml

```diff
@@ -1,35 +1,73 @@
 language: rust
 rust: nightly
 sudo: false
+services: docker
+cache: cargo
 
 matrix:
   include:
-    - env: TARGET=x86_64-unknown-linux-gnu NO_ADD=1
+    # Linux
+    - env: TARGET=aarch64-unknown-linux-gnu
+    - env: TARGET=arm-unknown-linux-gnueabi
+    - env: TARGET=arm-unknown-linux-musleabi
+    - env: TARGET=armv7-unknown-linux-gnueabihf
+    - env: TARGET=armv7-unknown-linux-musleabihf
+    - env: TARGET=i586-unknown-linux-gnu
+      addons:
+        apt:
+          packages:
+            - gcc-multilib
     - env: TARGET=i686-unknown-linux-gnu
       addons:
         apt:
           packages:
             - gcc-multilib
-      after_success:
-        - travis-cargo doc-upload
-    - os: osx
-      env: TARGET=x86_64-apple-darwin NO_ADD=1
-    - os: osx
-      env: TARGET=i686-apple-darwin
-    - env: TARGET=x86_64-unknown-linux-gnu NO_ADD=1
+    - env: TARGET=i686-unknown-linux-musl
+    - env: TARGET=mips-unknown-linux-gnu
+    - env: TARGET=mips64-unknown-linux-gnuabi64
+    - env: TARGET=mips64el-unknown-linux-gnuabi64
+    - env: TARGET=mipsel-unknown-linux-gnu
+    - env: TARGET=powerpc-unknown-linux-gnu
+    - env: TARGET=powerpc64-unknown-linux-gnu
+    - env: TARGET=powerpc64le-unknown-linux-gnu
+    - env: TARGET=sparc64-unknown-linux-gnu
+    - env: TARGET=x86_64-unknown-linux-gnu
+    - env: TARGET=x86_64-unknown-linux-musl
+
+    # Android
+    - env: TARGET=aarch64-linux-android
+    - env: TARGET=arm-linux-androideabi
+    - env: TARGET=armv7-linux-androideabi
+    - env: TARGET=x86_64-linux-android
+
+    # OSX
+    - env: TARGET=i686-apple-darwin
+      os: osx
+      osx_image: xcode9.2
+    - env: TARGET=x86_64-apple-darwin
+      os: osx
+      osx_image: xcode9.2
+
+    # Emscripten
+    - env: TARGET=asmjs-unknown-emscripten
+    - env: TARGET=wasm32-unknown-emscripten
+
+    - env: DOCUMENTATION
       install: pip install 'travis-cargo<0.2' --user && export PATH=$HOME/.local/bin:$PATH
       script: cargo doc
       after_success:
         - travis-cargo --only nightly doc-upload
 
-script:
-  - if [ -z "$NO_ADD" ]; then rustup target add $TARGET; fi
-  - cargo run --target $TARGET --manifest-path systest/Cargo.toml
-  - cargo build --target $TARGET
-  - cargo build --target $TARGET --features profiling
-  - cargo build --target $TARGET --features debug
-  - cargo build --target $TARGET --features 'debug profiling'
-  - cargo test --target $TARGET
+  allow_failures:
+    - env: TARGET=arm-linux-androideabi
+    - env: TARGET=armv7-linux-androideabi
+    - env: TARGET=arm-unknown-linux-musleabi
+    - env: TARGET=armv7-unknown-linux-musleabihf
+    - env: TARGET=sparc64-unknown-linux-gnu
+    - env: TARGET=asmjs-unknown-emscripten
+    - env: TARGET=wasm32-unknown-emscripten
```

Comment:
I think the emscripten targets can probably be safely removed, I don't think we'd ever want to compile jemalloc for those platforms

Reply:
Done